src/frontend/src/scheduler/distributed/query_manager.rs

```diff
@@ -132,6 +144,32 @@ impl QueryResultFetcher {
             yield DataChunk::from_protobuf(response?.get_record_batch()?)?;
         }
     }
+
+    fn run(self) -> BoxedDataChunkStream {
+        Box::pin(self.run_inner())
+    }
+
+    // #[try_stream(ok = DataChunk, error = RwError)]
+    async fn run_wrapper(self, shutdown_rx: Receiver<u8>) -> impl DataChunkStream {
+        let mut shutdown_rx = shutdown_rx;
+        let mut stream = self.run();
+        stream! {
```

Comment:
Does
```rust
self.run_inner().take_until(shutdown_rx)
```
work for you?

Reply:
I‘ll try

Reply:
This take_until works very well!

But there brings another question: I can only terminate the stream early, but not yield Error inside the Stream. This means even if the shutdown_rx trigger, the calller who consume can not know error happend but just know the stream stopped. 

TakeUntil has a method named `take_result`, so the solution may be: after consume the stream, use `take_result` to check stopping future resolved or not. If yes, extract the error inside it and returned. But now we are using BoxedStream or impl Stream every where, now all need to change to `TakeUntil<'_>` 😢